src/utilities.jl

```diff
@@ -32,19 +32,19 @@ Convert raster data between different formats.
 The output dataset.
 """
 function unsafe_gdaltranslate(
-        dataset::Dataset,
+        dataset::T,
         options = String[];
         dest = "/vsimem/tmp"
-    )
+    ) where {T<:AbstractDataset}
     options = GDAL.gdaltranslateoptionsnew(options, C_NULL)
     usage_error = Ref{Cint}()
     result = GDAL.gdaltranslate(dest, dataset.ptr, options, usage_error)
     GDAL.gdaltranslateoptionsfree(options)
-    return Dataset(result)
+    return T(result)
```

Comment:
This should still be `Dataset(result)` rather than `T(result)` (see https://github.com/yeesian/ArchGDAL.jl/pull/167#issuecomment-765704084 for context).